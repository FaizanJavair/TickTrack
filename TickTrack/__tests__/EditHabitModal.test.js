import React from "react";
import renderer from "react-test-renderer";
import EditHabitModal from "../modals/editHabitModal";

habit = {
  id: 1,
  name: "Drinking Water",
  color: "#414a4c",
  priority: "High",
  priorityValue: "2",
};

it("List Modal renders correctly", () => {
  const tree = renderer.create(<EditHabitModal habit={habit} />).toJSON();
  expect(tree).toMatchSnapshot();
});
