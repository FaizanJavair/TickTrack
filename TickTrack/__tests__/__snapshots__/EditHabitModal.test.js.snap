// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Modal renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "left": "7%",
        "opacity": 1,
        "position": "absolute",
        "top": "7%",
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      {
        "alignSelf": "stretch",
        "marginHorizontal": 40,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 30,
          "fontWeight": "600",
          "marginBottom": "5%",
          "textAlign": "center",
        }
      }
    >
      Edit Your Habit!
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholderTextColor="gray"
      style={
        {
          "borderColor": "#dadae8",
          "borderRadius": 12,
          "borderStyle": "solid",
          "borderWidth": 1.5,
          "height": 50,
          "paddingHorizontal": "2%",
        }
      }
      value="Drinking Water"
    />
    <View
      style={
        [
          {
            "width": "100%",
          },
          {
            "zIndex": 5000,
          },
          {},
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#dadae8",
            "borderRadius": 12,
            "borderStyle": "solid",
            "borderWidth": 1.5,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": "5%",
            "minHeight": 50,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 3,
            "width": "100%",
            "zIndex": 5000,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#000",
                "flex": 1,
              },
              {},
              false,
              {},
            ]
          }
        >
          Select an item
        </Text>
        <View
          style={
            [
              {
                "marginLeft": 10,
              },
              {},
            ]
          }
        >
          <Image
            source={1}
            style={
              [
                {
                  "height": 20,
                  "width": 20,
                },
                {},
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#414a4c",
          "borderRadius": 12,
          "height": 50,
          "justifyContent": "center",
          "marginTop": "6%",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 15,
            "fontWeight": "600",
          }
        }
      >
        Update
      </Text>
    </View>
  </View>
</View>
`;
